python/sglang/srt/entrypoints/openai/logprobs_processor.py

```diff
@@ -0,0 +1,144 @@
+from __future__ import annotations
+
+from typing import Any, Dict, List, Optional, Sequence
+
+from sglang.srt.entrypoints.openai.protocol import (
+    ChatCompletionTokenLogprob,
+    ChoiceLogprobs,
+    LogProbs,
+    TopLogprob,
+)
+
+
+class LogProbsProcessor:
+    """Factory converting raw token/top-logprobs to OpenAI-style ChoiceLogprobs."""
+
+    @classmethod
+    def from_response(cls, ret_item: Dict[str, Any]) -> ChoiceLogprobs:
```

Comment:
is `from_chat_completions` a better name?

Can we also add docstring for the function, so we know it is for chat completions?